tests/cpp-runtime/opencl/clml_memory_planner.cc

```diff
@@ -0,0 +1,481 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#include <gtest/gtest.h>
+#include <tvm/runtime/container/optional.h>
+
+#if defined(TVM_GRAPH_EXECUTOR_CLML)
+#include "../src/runtime/contrib/clml/clml_memory_planner.h"
+#include "../src/runtime/contrib/clml/clml_runtime.h"
+#include "../src/runtime/opencl/opencl_common.h"
+#include "../src/runtime/texture.h"
+
+using namespace tvm::runtime;
+using namespace tvm::runtime::cl;
+
+void InitMemoryPlan(tvm::runtime::contrib::CachedLayer& layer) {
```

Comment:
I saw that in class `CLMLRuntime` you have almost the same methods. Probably it is better to test them? You can change `private` to `protected` and inherit `CLMLRuntime` class in your test class. It should be like a test wrapper above `CLMLRuntime` class.

Reply:
Only ```InitMemoryPlan``` can be reused but ```PlanMemory```  has dependency with ```JSONRuntime``` nodes. initializing CLMLRuntime here requires Json graph and it's dependents. Hence I tried isolating the test environment within "CahcedLayer``` object. Let me see how much I can reuse from CLMLRuntime.